ochap-authentication/src/main/java/com/example/ochapauthentication/routes/AuthenticationController.java

```diff
@@ -0,0 +1,141 @@
+package com.example.ochapauthentication.routes;
+
+import com.example.ochapauthentication.commands.AccountRegisterCommand;
+import com.example.ochapauthentication.commands.AuthLoginCommand;
+import com.example.ochapauthentication.dto.AccountInfoDTO;
+import com.example.ochapauthentication.dto.ErrorDTO;
+import com.example.ochapauthentication.dto.ErrorsDTO;
+import com.example.ochapauthentication.dto.TokenDTO;
+import com.example.ochapauthentication.entities.Role;
+import com.example.ochapauthentication.entities.User;
+import com.example.ochapauthentication.jwt.JwtTokenUtil;
+import com.example.ochapauthentication.repository.RoleDAO;
+import com.example.ochapauthentication.repository.UserDAO;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+@Controller
+public class AuthenticationController {
+
+    final UserDAO userRepository;
+    final RoleDAO roleRepository;
+    final ObjectMapper objectMapper;
+
+    final int MIN_USERNAME_LENGTH = 3;
+    final int MIN_PASSWORD_LENGTH = 8;
+
+    @Autowired
+    AuthenticationController(UserDAO userRepository, RoleDAO roleRepository){
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @Autowired
+    private JwtTokenUtil jwt;
+
+    @PostMapping("/auth/login")
+    @ResponseBody
+    ResponseEntity<String> loginAutentication(@RequestBody AuthLoginCommand credentials) throws JsonProcessingException {
```

Comment:
```suggestion
    ResponseEntity<Object> loginAutentication(@RequestBody AuthLoginCommand credentials) throws JsonProcessingException {
```

Why not let spring to the marshalling for you?